### PR TITLE
Turn plot into box

### PR DESCRIPTION
**Pull request by birdsarah**
Part of #4164 

This PR:
* re-implements resize based on document resize and solver, not js on plot
* makes a toolbar a widget

ToDo moved to #4164

### CODE REVIEW

**Comment by birdsarah** [comment]
ping @fpliger @rothnic - this PR is not ready overall but there's one thing I would appreciate your eyes on: can you please check the small tweaks I made to the charts tests and chart.py. I basically just changed what looked sensible till they passed, but I'm not sure I did it right. Plot no longer has tools it has a Toolbar object.

**Comment by birdsarah** [comment]
ping @bryevdv 

You may want to start reviewing this. Please note that it is a PR against another branch not master. Hopefully this will make it slightly less work to review.

There is still more work to do, but there's some meaningful progress in here to review.

**Comment by bryevdv** [comment]
I ran the examples, looks nice so far, will look at code more closely later today.

File: bokeh/models/plots.py [hunk 1]
```diff
@@ -552,6 +561,42 @@ def border_fill(self, color):
             """)
         self.border_fill_color = color
 
+    @property
+    def logo(self):
+        warnings.warn(
+            """
+            Plot property 'logo' was deprecated in Bokeh 0.12.0 and will be removed.
+            User 'toolbar.logo' instead.
```
Comment (bryevdv) [inline bokeh/models/plots.py]: "User" -> "Use" also `Plot.toolbar.logo` more specific

File: bokeh/models/plots.py [hunk 2]
```diff
@@ -552,6 +561,42 @@ def border_fill(self, color):
...
+
+    @tools.setter
+    def tools(self, tools):
+        warnings.warn(
+            """
+            Plot property 'tools' was deprecated in Bokeh 0.12.0 and will be removed.
+            User 'toolbar.tools' instead.
+            """)
+        self.toolbar.tools = tools
+
```
Comment (bryevdv) [inline bokeh/models/plots.py]: same comment on all these

**Comment by birdsarah** [comment]
This is ready for review. I haven't made the changes Bryan made and I have a bunch of smaller ToDos that I've already noted above, but the meat of this is here. It will be much easier to review all the examples when the layout deprecation is done - that is happening here: #4333.

**Comment by birdsarah** [comment]
superseded: #4343

**Pull request closed without merging**